tests/standalone_tests/test_cleanup_after_algorithm_execution.py

```diff
@@ -0,0 +1,389 @@
+import asyncio
```

Comment:
I have tried some things/improvements in [this branch](https://github.com/madgik/MIP-Engine/tree/fix/MIP-275/suggestions), you can check it out and use what you think is useful.

In that branch I use mocking for the node_registry, controller_config and cdes, instead of using env variables.
This is done so that we can have more than one `controllers` in the future with different configurations.

In general I think 2 more tests should be added that would fail with the current implementation.
1) A test where the node process is killed but not the rabbitmq. The `node_id` remains in the node_registry.
2) A test where the node registry doesn't have one of the nodes that need cleanup. It's added later (the node comes back) and the cleanup finishes then.


It is also needed to deploy a controller with the whole stack so that dev_env tests can be completely replaced but it should be done with quart as well. If the normal deployment process is not followed, I think mocking is a safer approach and easier to change.

For example adding a test where a node is removed from the node_registry (connection lost) and then comes back and cleanup continues. This wouldn't work currently.

Reply:
added one more test that brings down only the (celery) node service and then brings it back up after the cleanup loop has started.
In the 2 tests that check if the cleanup loops correctly removes leftovers from the db of α node that goes down and then is brought βαψκ up (`test_cleanup_rabbitmq_down_algorithm_execution`,` test_cleanup_node_service_down_algorithm_execution`), if you check the code in the test you ll see that it is guaranteed that the node is removed from the node registry before it continues